lib/src/utils/l10n.dart

```diff
@@ -52,3 +53,50 @@ class _LocaleObserver extends WidgetsBindingObserver {
     _didChangeLocales(locales);
   }
 }
+
+/// Takes a localization function (from lib/l10n/l10n.dart) with a single placeholder
+/// and replaces the placeholder with the given widget, surrounded by Text widgets if necessary.
+///
+/// For example:
+///
+/// ```dart
+/// String translationFun(String name) {
+///   return 'Hello, ${name}!';
+/// }
+///
+/// // returns [Text('Hello, '), MyFancyWidget('Magnus'), Text('!')]
+/// replaceL10nPlaceholderWithWidget(translationFun, MyFancyWidget('Magnus'));
+/// ```
+///
+/// @param l10nWithReplacement The localization function corresponding to a localized string with a placeholder.
+/// @param widget The widget to replace the placeholder with.
+/// @param testStyle The style to apply to the Text widgets.
+///
+/// If the placeholder is not present in the localized string,
+/// will return a single Text widget with the localized string.
+/// If the placeholder is at the beginning or end of the localized string,
+/// will return a list with a single Text widget and the widget.
+/// Otherwise, will return a list with three Text widgets:
+/// the first with the text before the placeholder, the second with the widget,
+/// and the third with the text after the placeholder.
+IList<Widget> replaceL10nPlaceholderWithWidget<T extends Widget>(
```

Comment:
Instead of an `List<Widget>`, the idiomatic way to handle this in flutter is the [Text.rich](https://api.flutter.dev/flutter/widgets/Text/Text.rich.html) constructor.

See [WidgetSpan](https://api.flutter.dev/flutter/widgets/WidgetSpan-class.html) to embed a widget inline inside text.

Reply:
thanks for the hint!